presto-main-base/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1272,6 +1299,248 @@ protected Scope visitLateral(Lateral node, Optional<Scope> scope)
+                    }
+                    passedArguments.put(argumentSpecification.getName(), analyzeArgument(argumentSpecification, argument));
+                }
+                // apply defaults for not specified arguments
+                for (Map.Entry<String, ArgumentSpecification> entry : argumentSpecificationsByName.entrySet()) {
+                    ArgumentSpecification argumentSpecification = entry.getValue();
+                    passedArguments.put(argumentSpecification.getName(), analyzeDefault(argumentSpecification, errorLocation));
+                }
+            }
+            else {
+                for (int i = 0; i < arguments.size(); i++) {
+                    TableFunctionArgument argument = arguments.get(i);
+                    ArgumentSpecification argumentSpecification = argumentSpecifications.get(i); // TODO args passed positionally - can one only pass some prefix of args?
+                    passedArguments.put(argumentSpecification.getName(), analyzeArgument(argumentSpecification, argument));
+                }
+                // apply defaults for not specified arguments
+                for (int i = arguments.size(); i < argumentSpecifications.size(); i++) {
+                    ArgumentSpecification argumentSpecification = argumentSpecifications.get(i);
+                    passedArguments.put(argumentSpecification.getName(), analyzeDefault(argumentSpecification, errorLocation));
+                }
+            }
+
+            return passedArguments.buildOrThrow();
+        }
+
+        private Argument analyzeArgument(ArgumentSpecification argumentSpecification, TableFunctionArgument argument)
+        {
+            String actualType;
+            if (argument.getValue() instanceof TableFunctionTableArgument) {
+                actualType = "table";
+            }
+            else if (argument.getValue() instanceof TableFunctionDescriptorArgument) {
+                actualType = "descriptor";
+            }
+            else if (argument.getValue() instanceof Expression) {
+                actualType = "expression";
+            }
+            else {
+                throw new SemanticException(INVALID_FUNCTION_ARGUMENT, argument, "Unexpected table function argument type: ", argument.getClass().getSimpleName());
+            }
+
+            if (argumentSpecification instanceof TableArgumentSpecification) {
+                if (!(argument.getValue() instanceof TableFunctionTableArgument)) {
+                    if (argument.getValue() instanceof FunctionCall) {
+                        // probably an attempt to pass a table function call, which is not supported, and was parsed as a function call
+                        throw new SemanticException(NOT_SUPPORTED, argument, "Invalid table argument %s. Table functions are not allowed as table function arguments", argumentSpecification.getName());
```

Comment:
What if its not a direct argument but an argument which is one level deep? like

**tvf ( max(tvf2))**?